apps/chat/tasks.py

```diff
@@ -22,6 +22,30 @@
 STATUSES_FOR_COMPLETE_CHATS = [SessionStatus.PENDING_REVIEW, SessionStatus.COMPLETE, SessionStatus.UNKNOWN]
 
 
+def _get_latest_sessions_for_participants(
+    participant_chat_ids: list, experiment_public_id: UUID
+) -> list[ExperimentSession]:
+    latest_session_id = (
+        ExperimentSession.objects.filter(experiment__public_id=experiment_public_id, participant=OuterRef("pk"))
+        .order_by("-created_at")
+        .values("id")[:1]
+    )
+
+    latest_participant_session_ids = (
+        Participant.objects.filter(
+            experimentsession__experiment__public_id=experiment_public_id, external_chat_id__in=participant_chat_ids
+        )
+        .annotate(latest_session_id=Subquery(latest_session_id))
+        .values("latest_session_id")
+    )
+
+    return (
+        ExperimentSession.objects.filter(id__in=Subquery(latest_participant_session_ids))
+        .prefetch_related("experiment")
+        .all()
+    )
+
+
 @shared_task
 def send_bot_message_to_users(message: str, chat_ids: list[str], is_bot_instruction: bool, experiment_public_id: UUID):
```

Comment:
probably for a separate PR but why does this need to support multiple chat IDs? As far as I can tell we only lever pass in one

Reply:
So this is task was originally built to support both tools and be used for manual scheduled messages. There are one or two cases where multiple chat IDs were used. This will be made obsolete by the scheduled message changes